File: src/main/java/com/mundial2026/service/EquipoService.java
package com.mundial2026.service;

import java.util.List;

import com.mundial2026.model.entity.Equipo;

public interface EquipoService {

    List<Equipo> obtenerTodos();
    Equipo obtenerPorId(Long id);
    Equipo guardar(Equipo equipo);
    void eliminar(Long id);
}

File: frontend/src/app/services/equipo.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Equipo } from '../models/equipo.model';

@Injectable({ providedIn: 'root' })
export class EquipoService {

  private apiUrl = 'http://localhost:8080/api/equipos';

  constructor(private http: HttpClient) {}

  obtenerTodos(): Observable<Equipo[]> {
    return this.http.get<Equipo[]>(this.apiUrl);
  }

  obtenerPorId(id: number): Observable<Equipo> {
    return this.http.get<Equipo>(`${this.apiUrl}/${id}`);
  }
}